File: BOJ/3023_1.cpp
// 마술사 이민혁
// https://www.acmicpc.net/problem/3023
#include <iostream>
using namespace std;
const int MAX = 100 + 1;
char board[MAX][MAX];
int r, c;
void right(int sr, int sc)
{
    for (int i = 0; i < r; i++)
    {
        for (int j = 0; j < c; j++)
        {
            board[i][(2 * c - 1) - j] = board[i][j];
        }
    }
}
void down(int sr, int sc)
{
    for (int i = 0; i < r; i++)
    {
        for (int j = 0; j < c; j++)
        {
            board[(2 * r - 1) - i][j] = board[i][j];
        }
    }
}
void dig(int sr, int sc)
{
    for (int i = 0; i < r; i++)
    {
        for (int j = 0; j < c; j++)
        {
            board[(2 * r - 1) - i][(2 * c - 1) - j] = board[i][j];
        }
    }
}
int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);

    cin >> r >> c;
    for (int i = 0; i < r; i++)
        for (int j = 0; j < c; j++)
            cin >> board[i][j];

    int sr = 0, sc = 0;
    right(sr, sc + c);
    down(sr + r, sc);
    dig(sr + r, sc + c);

    int er, ec;
    cin >> er >> ec;
    if (board[er - 1][ec - 1] == '.')
        board[er - 1][ec - 1] = '#';
    else
        board[er - 1][ec - 1] = '.';

    for (int i = 0; i < r * 2; i++)
    {
        for (int j = 0; j < c * 2; j++)
            cout << board[i][j];
        cout << '\n';
    }

    return 0;
}

File: BOJ/3023_1.py
'''
    마술사 이민혁
    https://www.acmicpc.net/problem/3023
'''
R, C = map(int, input().split())
board = []
for _ in range(R):
    line = input()
    board.append(line + line[::-1])  # 대칭
A, B = map(int, input().split())
A -= 1
B -= 1

for i in range(2 * R):
    index = i
    if i >= R:
        index = 2 * R - i - 1
    if i == A:
        s = board[index]
        '''
        if s[B] == '.':
            temp = '#'
        else:
            temp = '.'
        '''
        temp = '#' if s[B] == '.' else '.'
        print(s[:B] + temp + s[B+1:])
        continue
    print(board[index])
'''
    참고
    https://blog.wonkyunglee.io/3
    파이썬 string은 불변 -> 복사해야함
    https://hashcode.co.kr/questions/6444/%ED%8C%8C%EC%9D%B4%EC%8D%AC%EC%97%90%EC%84%9C-str-object-does-not-support-item-deletion%EB%9D%BC%EB%8A%94-%EC%97%90%EB%9F%AC%EA%B0%80-%EB%82%AC%EC%8A%B5%EB%8B%88%EB%8B%A4
    https://www.acmicpc.net/source/15531193
'''
